packages/notebook/src/widget.ts

```diff
@@ -1046,14 +1050,24 @@ export class StaticNotebook extends WindowedList<NotebookViewModel> {
     const isContentVisibility =
       this._notebookConfig.windowingMode === 'contentVisibility';
 
-    cell.toggleClass('jp-content-visibility', isContentVisibility);
-
-    if (isContentVisibility) {
-      const estHeight = this._viewModel.estimateWidgetSize(index);
-      cell.node.style.containIntrinsicSize = `auto ${estHeight}px`;
-    } else {
+    if (!isContentVisibility) {
       cell.node.style.removeProperty('contain-intrinsic-size');
+      cell.toggleClass('jp-content-visibility', false);
+      return;
     }
+
+    const estHeight = this._viewModel.estimateWidgetSize(index);
+
+    requestAnimationFrame(() => {
+      cell.toggleClass('jp-content-visibility', true);
+      cell.node.style.containIntrinsicSize = `auto ${estHeight}px`;
+
+      // Update height estimate in the view model
+      this.viewModel.setEstimatedWidgetSize(
+        cell.model.id,
+        cell.node.getBoundingClientRect().height
+      );
```

Comment:
Do we need this here if it is covered by `_runOnIdleTime`? It would defeat the purpose of running on idle if we do that on every cell update?